CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -99,7 +202,8 @@ public static function cache($userID, $type = CRM_Core_Permission::VIEW, $force
     }
 
     if (!$force) {
-      if (!empty($_processed[$userID])) {
+      // skip if already calculated
+      if (!empty($_processed[$type][$userID])) {
```

Comment:
won't change now, since this is also the existing approach. But I added a FIXME comment